### PR TITLE
Add a unit test for editing annotations

### PR DESCRIPTION
**Pull request by Toromtomtom**
This adds a unit test for the annotation editing introduced in https://github.com/koreader/koreader-base/pull/1327.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1331)

### CODE REVIEW

File: spec/unit/mupdf_spec.lua [hunk 1]
```diff
@@ -137,6 +139,29 @@ describe("mupdf module", function()
...
+            assert.is_not_nil(page)
+            local annot = page:getMarkupAnnotation(ffi.new("float[8]", {
+                 70,  930,
+                510,  930,
+                510,  970,
+                 70,  970 }),
+                1)
+            page:updateMarkupAnnotation(annot, "annotation contents")
+            page:close()
+            doc:writeDocument(simple_pdf_out_annotated)
```
Comment (NiLuJe) [inline spec/unit/mupdf_spec.lua:156]: This is probably missing a `doc:close()` (As does the existing test above, apparently :/).

File: spec/unit/mupdf_spec.lua [hunk 2]
```diff
@@ -137,6 +139,29 @@ describe("mupdf module", function()
...
+            local out_f = io.open(simple_pdf_out_annotated, "r")
+            out_f:read("*a")
+            out_f:close()
```
Comment (NiLuJe) [inline spec/unit/mupdf_spec.lua:159]: Also applies to the existing test above: I have no idea what this is attempting to test :D.

File: spec/unit/mupdf_spec.lua (hunk 1 above)
Comment (Toromtomtom) [reply spec/unit/mupdf_spec.lua:155]: Thanks, I added the `doc:close()` to both tests.

File: spec/unit/mupdf_spec.lua (hunk 2 above)
Comment (Toromtomtom) [reply spec/unit/mupdf_spec.lua]: Thanks, removed it in both tests.

**Review by NiLuJe — APPROVED** [review]
Thanks!

**Review by Frenzie — APPROVED** [review]

**Pull request merged**